src/dialogs/confirmation-dialog.ts

```diff
@@ -0,0 +1,82 @@
+import { LitElement, html, TemplateResult, CSSResultGroup, css } from "lit";
+import { customElement, state } from "lit/decorators.js";
+import "@material/mwc-button";
+import "@material/mwc-dialog";
+import { ConfirmationDialogParams } from ".";
+import { classMap } from "lit/directives/class-map.js";
+
+@customElement("esphome-confirmation-dialog")
+class ESPHomeConfirmationDialog extends LitElement {
+  @state() private _params?: ConfirmationDialogParams;
+  @state() private _resolve?: (value: boolean | PromiseLike<boolean>) => void;
+
+  public async showDialog(
+    params: ConfirmationDialogParams,
+    resolve: (value: boolean | PromiseLike<boolean>) => void
+  ): Promise<void> {
+    this._params = params;
+    this._resolve = resolve;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params) {
+      return html``;
+    }
+    return html`
+      <mwc-dialog
+        .heading=${this._params.title || ""}
+        @closed=${this._handleClose}
+        open
+      >
+        ${this._params.text && html`<div>${this._params.text}</div>`}
```

Comment:
This is not React where boolean values are filtered out. Use ternary statements.
```suggestion
        ${this._params.text ? html`<div>${this._params.text}</div>` : ""}
```

Reply:
oops